### DISCUSSION TITLE
Add support for fan entities

### DISCUSSION THREAD

**Discussion by mouth4war** [jseidl/magic-areas#198] [Ideas / Feature Requests]
Fan entities should also be turned on and off based on states

**Comment by jseidl** [comment]
So the idea is to use a [Generic thermostat](https://www.home-assistant.io/integrations/generic_thermostat/) attached to the fan and a temperature sensor (say `sensor.area_temperature_living_room`), then climate groups will handle the rest

**Reply by mouth4war** [reply]
Multiple fans per room means creating many more generic thermostats just for the purpose of magic areas turning them on and off.

Please reconsider adding fan support so air purifiers, ceiling fans and ventilation fans can be handled.

**Comment by jseidl** [comment]
Makes sense for ventilation fans, will need to create a Fan group. 

For ceiling fans, for me at least, makes sense have a generic thermostat because I don't want them turning on when it's cold.

**Comment by jseidl** [comment]
Hey @mouth4war, years later, I'm working on https://github.com/jseidl/hass-magic_areas/pull/492 which partially does what you want, with the exception that you can only have one per area.

Fan groups will grab ALL fans in an area and group them. You will be able to tell them to track ONE aggregate (temperature/humidity/co2 etc) and set a setpoint which the fans will be ON when over it, OFF when under.

This allows one to treat ALL fans in an area as ONE of:
* comfort fans can be tied to temperature
* air purifiers to vocs
* exhaust fans to co2 
* etc...

Sorry for the caps, just wanted to be very clear.

In the future I plan an update where one could create multiple fan groups. Once I figure this out probably there will be the same for light groups, instead of just those 4.

**Comment by jseidl** [comment]
Thou shall have it https://github.com/jseidl/hass-magic_areas/pull/492
